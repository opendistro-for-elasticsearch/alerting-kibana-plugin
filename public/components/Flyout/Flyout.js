/*
 *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import React from 'react';
import PropTypes from 'prop-types';
import { EuiFlyout, EuiFlyoutBody, EuiFlyoutHeader, EuiFlyoutFooter } from '@elastic/eui';

import Flyouts from './flyouts';

const getFlyoutData = ({ type, payload }) => {
  const flyout = Flyouts[type];
  if (!flyout || typeof flyout !== 'function') return null;
  return flyout(payload);
};

const propTypes = {
  flyout: PropTypes.object,
  onClose: PropTypes.func.isRequired,
};

const Flyout = ({ flyout, onClose }) => {
  if (!flyout) return null;
  const flyoutData = getFlyoutData(flyout);
  if (!flyoutData) return null;
  const {
    flyoutProps = {},
    headerProps = {},
    bodyProps = {},
    footerProps = {},
    header = null,
    body = null,
    footer = null,
  } = flyoutData;

  const flyoutHeader = header && <EuiFlyoutHeader {...headerProps}>{header}</EuiFlyoutHeader>;
  const flyoutBody = body && (
    <EuiFlyoutBody onClose={onClose} {...bodyProps}>
      {body}
    </EuiFlyoutBody>
  );
  const flyoutFooter = footer && <EuiFlyoutFooter {...footerProps}>{footer}</EuiFlyoutFooter>;

  return (
    <EuiFlyout onClose={onClose} {...flyoutProps}>
      {flyoutHeader}
      {flyoutBody}
      {flyoutFooter}
    </EuiFlyout>
  );
};

Flyout.propTypes = propTypes;

export default Flyout;
