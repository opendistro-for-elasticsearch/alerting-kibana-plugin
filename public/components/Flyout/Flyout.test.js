/*
 *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import React from 'react';
import { shallow } from 'enzyme';

import Flyout from './Flyout';
import Flyouts from './flyouts';
jest.unmock('./flyouts');

describe('Flyout', () => {
  test('renders', () => {
    const wrapper = shallow(
      <Flyout flyout={{ type: 'message', payload: null }} onClose={jest.fn()} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('renders null if no flyout', () => {
    const wrapper = shallow(
      <Flyout flyout={{ type: 'definitely no flyout', payload: null }} onClose={jest.fn()} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('defaults if bad flyout data', () => {
    Flyouts.message = jest.fn(() => ({}));
    const wrapper = shallow(
      <Flyout flyout={{ type: 'message', payload: null }} onClose={jest.fn()} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
