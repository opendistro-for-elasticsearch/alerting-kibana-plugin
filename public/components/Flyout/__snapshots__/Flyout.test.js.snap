// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flyout defaults if bad flyout data 1`] = `
<EuiFlyout
  closeButtonAriaLabel="Closes this dialog"
  hideCloseButton={false}
  maxWidth={false}
  onClose={[MockFunction]}
  ownFocus={false}
  size="m"
/>
`;

exports[`Flyout renders 1`] = `
<EuiFlyout
  aria-labelledby="messageFlyout"
  closeButtonAriaLabel="Closes this dialog"
  hideCloseButton={false}
  maxWidth={500}
  onClose={[MockFunction]}
  ownFocus={false}
  size="m"
>
  <EuiFlyoutHeader
    hasBorder={true}
  >
    <EuiTitle
      size="m"
      style={
        Object {
          "fontSize": "25px",
        }
      }
    >
      <h2>
        <strong>
          Message
        </strong>
      </h2>
    </EuiTitle>
  </EuiFlyoutHeader>
  <EuiFlyoutBody>
    <EuiText
      style={
        Object {
          "fontSize": "14px",
        }
      }
    >
      <p>
        You have access to a "ctx" variable in your painless scripts and action mustache templates.
      </p>
      <h3>
        Learn More
      </h3>
      <ul>
        <li>
          <EuiLink
            color="primary"
            href="https://mustache.github.io/mustache.5.html"
            target="_blank"
            type="button"
          >
            HTML Templates with Mustache.js
          </EuiLink>
        </li>
      </ul>
    </EuiText>
  </EuiFlyoutBody>
</EuiFlyout>
`;

exports[`Flyout renders null if no flyout 1`] = `""`;
